packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -13,24 +15,68 @@ const BUTTER_SWAP_STATES = (() => ({
   Failed: 6,
 }))()
 
+// Cache relay IDs to avoid repeated API calls during polling
+// We use a module-level cache instead of storing in swap.metadata because:
+// 1. The relay ID is only available after transaction submission (not at quote time like chainflipSwapId)
+// 2. checkTradeStatus doesn't have a mechanism to update swap state during polling
+// 3. This approach is simpler and doesn't require changes to core types or polling logic
+const relayIdCache = new Map<string, number>()
```

Comment:
nitpick: relayer over relay not to confuse this with Relay, which my half-awake self may or may not have!

Reply:
Doh, missed that one: https://github.com/shapeshift/web/pull/9833/commits/24df3fb73b7c1ba6f366d13d81bb5f6b16753b54

Reply:
Oh wait, after more reading, I think this was meant to be `butterIdCache` 👀

Reply:
It's meant to be butter, sec